### PR TITLE
[WF] added Jenkinsfile and yaml file for daily builds

### PR DESCRIPTION
**Pull request by mbiarnes**
**Thank you for submitting this pull request**

https://github.com/kiegroup/kie-jenkins-scripts/pull/1203
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1884

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest</b> (using <i>this</i> e.g. <b>Jenkins retest this</b> optional but no longer required)
 
* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`
    
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by mbiarnes** [comment]
Jenkins retest this

**Review by mareknovotny — COMMENTED** [review]
is this ready for approval? i think it looks good to me.
wdyt @ginxo?

File: .ci/daily-config.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  - project: mbiarnes/droolsjbpm-build-bootstrap
+    build-command:
+      before:
+        current: |
+          ./script/release/eraseM2.sh ${{ env.MAVEN_DIR }}
+          ./script/release/update-versions-build-chain.sh ${{ env.KIE_VERSION }} community
+      current: |
+        ${{ env.MAVEN_COMMAND }}
+        ${{ env.WORKSPACE }}/clean-up.sh `pwd`
+  - project: mbiarnes/optaweb-vehicle-routing
```
Comment (Ginxo) [inline .ci/daily-config.yaml:26]: you can use
```
  - project: mbiarnes/optaweb-vehicle-routing
    skip: true
```

File: .ci/daily-config.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    build-command:
+      current: echo "Nothing to do for this project"
+  - project: mbiarnes/optaweb-employee-rostering
```
Comment (Ginxo) [inline .ci/daily-config.yaml:29]: you can use
```
  - project: mbiarnes/optaweb-employee-rostering
    skip: true
```

**Review by Ginxo — APPROVED** [review]
LGTM, just a minor thing.
Obviously `mbiarnes` should be replaced by `kiegroup` before merging.
Additionally... can't we remove deprecated scripts from `/script/folder`?

Good job!! it was not easy!

File: .ci/daily-config.yaml (hunk 1 above)
Comment (mbiarnes) [reply .ci/daily-config.yaml:26]: Done

File: .ci/daily-config.yaml (hunk 2 above)
Comment (mbiarnes) [reply .ci/daily-config.yaml:29]: Done

**Pull request merged**